st2client/st2client/models/core.py

```diff
@@ -274,11 +274,14 @@ def query(self, **kwargs):
             self.handle_error(response)
         items = response.json()
         instances = [self.resource.deserialize(item) for item in items]
-        return instances
+        if 'X-Total-Count' in response.headers:
```

Comment:
Yes, I have updated all the places, where query was called. However, unit tests are not happy, as we are mocking the response w/o headers.